api/src/main/java/org/apache/iceberg/FieldMetrics.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+
+import java.nio.ByteBuffer;
+
+/**
+ * Iceberg internally tracked field level metrics.
+ */
+public class FieldMetrics {
+  private final int id;
+  private final long valueCount;
```

Comment:
Do we need all these metrics that can be obtained from the footer? To me it feels more reasonable for this class to only contain metrics that is not covered by the footer metrics.

Reply:
Since I'll implement the same for ORC and Avro, and Avro currently doesn't have metrics support, in Avro we will populate the other fields. Ryan brought up a good idea of having a `ParquetFieldMetrics` to limit the access to non-nan fields, and I'll extend this class to implement `ParquetFieldMetrics`, and only initialize `ParquetFieldMetrics` in parquet related writers.